error_tracker/django/views.py

```diff
@@ -35,17 +36,30 @@ def view_list(request):
     :return:  rendered template
     """
     title = "App Error"
-    try:
-        page = int(request.GET.get('page', 1))
-    except:
-        page = 1
+    
+    query = request.GET.dict()
+    
     error = False
-    errors = model.get_exceptions_per_page(page_number=page)
+    errors = model.get_exceptions_per_page(**query)
```

Comment:
We need to set page_number if it's missing. And please add **kwargs in ModelMixin, for compatibility.

Reply:
Don't understand your meaning :
**on page_number** 
```
def get_exceptions_per_page(cls, **query):
        if 'page' in query:
            page_number = query['page']
            del query['page']
        else:
            page_number = 1
```

>And please add **kwargs in ModelMixin, for compatibility.

You mean replace in ModelMixin :
```
@classmethod
@abc.abstractmethod
def get_exceptions_per_page(cls, page_number=1):
```
with
```
@classmethod
@abc.abstractmethod
def get_exceptions_per_page(cls, **kwargs):
```

Reply:
We need to update model mixin as

```
@classmethod
@abc.abstractmethod
def get_exceptions_per_page(cls, page_number=1, **kwargs):


```



Now when we send GET query parameter as dict than we should do something like this


`model.get_exceptions_per_page(page_number=page, **query)
`

I know this will still break the compatibility, so we'll release 2.0.0 to make things fine.

Reply:
ok, 

i set in Mixin :
```
    @classmethod
    @abc.abstractmethod
    def get_exceptions_per_page(cls, page_number=1, **kwargs):
        """
        An object having these properties,
        has_next, next_num, has_prev, prev_num and items
        on the returned object like SQLAlchemy's Pagination
        :return: a paginated object
        """
        raise NotImplementedError
```

In models.py
```
    @classmethod
    def get_exceptions_per_page(cls, page_number=1, **query):
        if query and 'page' in query:
            page_number = query['page']
            del query['page']
            
        if not query:
            records = cls.objects.all().order_by('last_seen')
        else:
            records = cls.objects.filter(**query).order_by('last_seen')
```

Ok for you ?

Reply:
yes this works for me, but do you any side of using **query in filter? For example how it will handle the case of partial match.

Reply:
step by step :-). We will start with a full search and study an easy way to make partial search later..
(this later could be there quickly :-) )

Reply:
and the time is there :-):

Could be manager like this:
```
query = {"{}__contains".format(k): v for k, v in query.items()}
```